### PR TITLE
Add integer-based std_pow

### PR DESCRIPTION
**Pull request by cgyurgyik**

### CODE REVIEW

**Comment by cgyurgyik** [comment]
@rachitnigam Should this be moved to unsigned? Eventually I'll need to implement another for fixed point.

File: tests/correctness/exponent.futil [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import "primitives/std.lib";
+import "primitives/bitnum/math.futil";
+component main() -> () {
+  cells {
+    base = std_const(32,3);
+    exp = std_const(32,2);
+    b_x = std_mem_d1(32, 1, 1);
```
Comment (rachitnigam) [inline tests/correctness/exponent.futil:7]: I would call this "out" if this is the output memory. Also, would it make sense to make the inputs (base and exp) be read from other memories?

Comment (rachitnigam) [reply tests/correctness/exponent.futil]: `init` doesn't seem to use `1` as its done condition?

File: tests/correctness/exponent.futil [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    exp = std_mem_d1(32, 1, 1);
+    out = std_mem_d1(32, 1, 1);
+    std_exp0 = std_exp();
+  }
+  wires {
+    group init {
+      base.addr0 = 1'd0;
+      exp.addr0 = 1'd0;
+      init[done] = 1'd1;
+    }
```
Comment (cgyurgyik) [inline tests/correctness/exponent.futil:15]: I meant in the test here  @rachitnigam

Comment (rachitnigam) [reply tests/correctness/exponent.futil:15]: Yeah, this program is undefined. Using groups with done condition `1` within a seq doesn't work. `invoke` has a terrible interface—the inputs must be driven forever. This means that `init` should read the values those memories and saves them to registers. Next, you should use the registers to supply the value to the `invoke`.

**Comment by cgyurgyik** [comment]
This should probably be re-named to `std_pow` to follow classic naming convention.

**Comment by rachitnigam** [comment]
LGTM, merge whenever you're ready.

**Comment by cgyurgyik** [comment]
I also failed to mention that that the actual component was essentially pulled from @rachitnigam's Invoke PR, #300.

**Pull request merged**